### ISSUE TITLE
Android build issues

### ISSUE THREAD

**Issue by centsibly-admin** [plaid/react-native-plaid-link-sdk#124]
Hello!

I get an error while trying to run `react-native run-android` when using `"react-native-plaid-link-sdk": "3.1.2",`.

```
error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/getting-started.html#android-development-environment. Run CLI with --verbose flag for more details.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':react-native-plaid-link-sdk:generateDebugRFile'.
> Could not resolve all task dependencies for configuration ':react-native-plaid-link-sdk:debugRuntimeClasspath'.
   > Could not resolve com.facebook.react:react-native:+.
     Required by:
         project :react-native-plaid-link-sdk
      > Failed to list versions for com.facebook.react:react-native.
         > Unable to load Maven meta-data from https://artifactory.plaid.io/artifactory/link-sdk/com/facebook/react/react-native/maven-metadata.xml.
            > Could not get resource 'https://artifactory.plaid.io/artifactory/link-sdk/com/facebook/react/react-native/maven-metadata.xml'.
               > Could not GET 'https://artifactory.plaid.io/artifactory/link-sdk/com/facebook/react/react-native/maven-metadata.xml'.
                  > Connect to artifactory.plaid.io:443 [artifactory.plaid.io/23.217.138.110, artifactory.plaid.io/23.202.231.169] failed: connect timed out

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

```

I traced this to the gradle build file in react-native-plaid-link-sdk (https://github.com/plaid/react-native-plaid-link-sdk/blob/f58ea644bdc52f9abac04aea661c57f0b5448b17/android/build.gradle#L11), and if I comment out the maven repository `artifcatory.plaid.io` I'm able to get `react-native run-android` working again.

```
allprojects {
    repositories {
        mavenLocal()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
        google()
        jcenter()
    //    maven {
    //        url "https://artifactory.plaid.io/artifactory/link-sdk"
    //    }
    }
}
```

It looks like maybe that artifactory repository was taken offline? Maybe people weren't aware it was still referenced here?

Thanks so much.

**Comment by zsweigart** [comment]
Thanks for the report.  We'll remove it

**Comment by zsweigart** [comment]
@centsibly-admin this will be updated in the next release https://github.com/plaid/react-native-plaid-link-sdk/pull/125

**Comment by centsibly-admin** [comment]
Great! Thanks so much.

**Comment by zsweigart** [comment]
Hi  @centsibly-admin we have released a new  version today, please let us know if the issue has been resolved for you.

**Comment by centsibly-admin** [comment]
Hey @zsweigart,

Thanks for pushing the fix.

I actually found that the old version I was using started working fine (3.1.2). However when I upgrade to 3.3.0 I get this error:

```
> Task :react-native-plaid-link-sdk:compileDebugKotlin FAILED
w: Detected multiple Kotlin daemon sessions at build/kotlin/sessions
...
e: /.../mobile/node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PlaidModule.kt: (151, 17): Unresolved reference: oauthNonce
```

**Comment by enriquezc** [comment]
Hello @zsweigart ,

I'm getting the same error message trying to install my app with new version (3.3.0) on an Android simulator. This error doesn't get reproduced on an iOS simulator. 

`> Task :react-native-plaid-link-sdk:compileDebugKotlin FAILED
135 actionable tasks: 2 executed, 133 up-to-date
e: /.../node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PlaidModule.kt: (151, 17): Unresolved reference: oauthNonce`

I'm hoping that you can let me know what version is safe to roll back to. Thank you for the help!

**Comment by ryanleesipes** [comment]
I'm seeing the same error as well, curious as to what a reasonable workaround would be?

**Comment by sebastianpaz** [comment]
@enriquezc I've been using 3.1.2 and works fine in both iOS and Android.
I'm also curious whether there is a workaround we can use to pull this through.

**Comment by zsweigart** [comment]
Hi @centsibly-admin we just released `3.3.2` which should fixe the `Unresolved reference: oauthNonce` issue

**Comment by zsweigart** [comment]
Hi @centsibly-admin  we have release `4.0.1` which has an updated API, please let us know if you are still having build issues.
